### ISSUE TITLE
Autochoice and Table filtering does not occur for strings with newline characters in them.

### ISSUE THREAD

**Issue by coderazzi** [coderazzi/tablefilter-swing#66]
(Original issue 20 created by None on 2011-11-01T10:58:47+00:00)

```
Steps to reproduce:
1. Initialize TableFilterHeaders for a jTable with AutoComplete and AutoChoice properties
set to true for a particular column
2. This column cells are rendered by TextAreaRenderer which accepts newline characters.
(eg string: &quot;Test \n String&quot;)
3. Check for filtering with characters that occur after the newline.(eg: &quot;*String*&quot;)

In the filter dropdown for this column, the characters appearing after the newline(&quot;\n&quot;)
does not appear, nor in the wildcard search.

When &quot;*\\n*String&quot; is specified, the filters work as expected, but is inconsistent.

Could the mentioning of escape characters be avoided and made as default to search
inclusive of characters occurring after the newline and other special characters?

Running v4.1.4 on WindowsXP.



```

Original issue reported on code.google.com by `karthikv88` on 2011-11-01 10:58:47

**Comment by coderazzi** [comment]
(Original issue 20 last updated on 2011-11-01T10:58:47+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by coderazzi** [comment]
(Original comment by {u'display_name': u'coderazzi coderazzi', u'account_id': u'557058:790eb198-2ae4-49b4-b920-8d654aa86b3f'} on 2011-11-09T11:33:03+00:00)

```
(No text was entered with this change)
```

Original issue reported on code.google.com by `coderazzi` on 2011-11-09 11:33:03

**Comment by coderazzi** [comment]
(Original comment by {u'display_name': u'coderazzi coderazzi', u'account_id': u'557058:790eb198-2ae4-49b4-b920-8d654aa86b3f'} on 2011-11-09T16:17:20+00:00)

```
The issue is easily solved on the Parser class, based on regular expressions, including
the Pattern.DOTALL flag (without side effects).

However, this raises other question: how the content appears on the filter editor with
new lines (see attached image). This would normally need the user to setup a specific
renderer (but then it would loose the opportunity to search text), or remove AutoChoices.Enabled
for that column.
```

Original issue reported on code.google.com by `coderazzi` on 2011-11-09 16:17:20
<hr>
 * *Attachment: [tf20.JPG](https://storage.googleapis.com/google-code-attachments/tablefilter-swing/issue-20/comment-2/tf20.JPG)*

**Comment by coderazzi** [comment]
(Original comment by {u'display_name': u'coderazzi coderazzi', u'account_id': u'557058:790eb198-2ae4-49b4-b920-8d654aa86b3f'} on 2011-11-10T16:20:06+00:00)

```
Fixed on 4.2.0
```

Original issue reported on code.google.com by `coderazzi` on 2011-11-10 16:20:06
